portfolio/src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -40,13 +41,32 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     JSONArray eventsArray = jsonFromRequest.getJSONArray("events");
     JSONArray tasksArray = jsonFromRequest.getJSONArray("tasks");
-    String workHoursStartTimeString = (String) jsonFromRequest.get("startTime");
-    String workHoursEndTimeString = (String) jsonFromRequest.get("endTime");
+    String workHoursStartTimeString = jsonFromRequest.getString("startTime");
+    String workHoursEndTimeString = jsonFromRequest.getString("endTime");
     Instant workHoursStartTime = Instant.parse(workHoursStartTimeString);
     Instant workHoursEndTime = Instant.parse(workHoursEndTimeString);
-
+    String algorithmTypeString = jsonFromRequest.getString("algorithmType");
     Collection<CalendarEvent> events = collectEventsFromJsonArray(eventsArray);
     Collection<Task> tasks = collectTasksFromJsonArray(tasksArray);
+
+    // These are null for initialization purposes. They will either be set or
+    // the program will thrown an error.
+    SchedulingAlgorithmType schedulingAlgorithmType = null;
+    BaseTaskScheduler algorithm = null;
+
+    // This will take the string representation of the algorithm that is passed
+    // in and match it to something in the enum under BaseTaskScheduler.
+    try {
```

Comment:
Consider using a helper method that returns an optional containing SchedulingAlgorithmType.

ex:

Optional<SchedulingAlgorithmType> schedulingAlgorithmTypeOptional =
    getSchedulingAlgorithmTypeOptional(...);
if (!schedulingAlgorithmTypeOptional.isPresent()) {
  // Return a 400 response to the client and return the method.
}

Optional<BaseTaskScheduler> algorithmOptional = getAlgorithmOptional(...);
if (!algorithmOptional.isPresent()) {
  // Same thing here
}

...

Reply:
I think we can skip the `schedulingAlgorithmType` variable too since it's only used here, and like James said just write a `getAlgorithmOptional(...)` method that takes in the String.

Reply:
Done.